api/v1alpha1/nodehealthcheck_webhook.go

```diff
@@ -74,76 +79,126 @@ func (r *NodeHealthCheck) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	nodehealthchecklog.Info("validate update", "name", nhc.Name)
 
 	// do the normal validation
-	if err := r.validate(); err != nil {
+	if err := nhc.validate(); err != nil {
 		return err
 	}
 
 	// during ongoing remediations, some updates are forbidden
-	if r.isRemediating() {
-		if updated, field := r.isRestrictedFieldUpdated(old.(*NodeHealthCheck)); updated {
+	if nhc.isRemediating() {
+		if updated, field := nhc.isRestrictedFieldUpdated(old.(*NodeHealthCheck)); updated {
 			return fmt.Errorf("%s update %s", field, OngoingRemediationError)
 		}
 	}
 	return nil
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
-func (r *NodeHealthCheck) ValidateDelete() error {
-	nodehealthchecklog.Info("validate delete", "name", r.Name)
-	if r.isRemediating() {
+func (nhc *NodeHealthCheck) ValidateDelete() error {
+	nodehealthchecklog.Info("validate delete", "name", nhc.Name)
+	if nhc.isRemediating() {
 		return fmt.Errorf("deletion %s", OngoingRemediationError)
 	}
 	return nil
 }
 
-func (r *NodeHealthCheck) validate() error {
-	aggregated := errors.NewAggregate([]error{r.validateMinHealthy(), r.validateSelector()})
+func (nhc *NodeHealthCheck) validate() error {
+	aggregated := errors.NewAggregate([]error{
+		nhc.validateMinHealthy(),
+		nhc.validateSelector(),
+		nhc.validateMutualRemediations(),
+		nhc.validateEscalatingRemediations(),
+	})
 
 	// everything else should have been covered by API server validation
 	// as defined by kubebuilder validation markers on the NHC struct.
 
 	return aggregated
 }
 
-func (r *NodeHealthCheck) validateMinHealthy() error {
+func (nhc *NodeHealthCheck) validateMinHealthy() error {
 	// Using Minimum kubebuilder marker for IntOrStr does not work (yet)
-	if r.Spec.MinHealthy == nil {
+	if nhc.Spec.MinHealthy == nil {
 		return fmt.Errorf("MinHealthy must not be empty")
 	}
-	if r.Spec.MinHealthy.Type == intstr.Int && r.Spec.MinHealthy.IntVal < 0 {
-		return fmt.Errorf("%s: %v", minHealthyError, r.Spec.MinHealthy)
+	if nhc.Spec.MinHealthy.Type == intstr.Int && nhc.Spec.MinHealthy.IntVal < 0 {
+		return fmt.Errorf("%s: %v", minHealthyError, nhc.Spec.MinHealthy)
 	}
 	return nil
 }
 
-func (r *NodeHealthCheck) validateSelector() error {
-	if _, err := metav1.LabelSelectorAsSelector(&r.Spec.Selector); err != nil {
+func (nhc *NodeHealthCheck) validateSelector() error {
+	if _, err := metav1.LabelSelectorAsSelector(&nhc.Spec.Selector); err != nil {
 		return fmt.Errorf("%s: %v", invalidSelectorError, err.Error())
 	}
 	return nil
 }
 
-func (r *NodeHealthCheck) isRestrictedFieldUpdated(old *NodeHealthCheck) (bool, string) {
+func (nhc *NodeHealthCheck) validateMutualRemediations() error {
+	if nhc.Spec.RemediationTemplate == nil && (nhc.Spec.EscalatingRemediations == nil || len(nhc.Spec.EscalatingRemediations) == 0) {
+		return fmt.Errorf(mandatoryRemediationError)
+	}
+	if !(nhc.Spec.RemediationTemplate == nil || nhc.Spec.EscalatingRemediations == nil) {
```

Comment:
Nit: IMO better to avoid negative condition for simple cases.
i.e replace:
```
!(nhc.Spec.RemediationTemplate == nil || nhc.Spec.EscalatingRemediations == nil)
```
with:
```
(nhc.Spec.RemediationTemplate != nil && nhc.Spec.EscalatingRemediations != nil)
```